pfam_experiment.py

```diff
@@ -46,186 +48,439 @@
+            batch_size=FLAGS.knn_batch_size,
+            train_samples=1,
+            shuffle_seed=FLAGS.knn_shuffle_seed,
+            sample_random_state=FLAGS.knn_sample_random_state))
+
+    for knn_train_samples in knn_train_samples_:
+
+        datum.update(
+            measure_nearest_neighbor_performance(
+                accuracy_label='test_knn_accuracy_untrained_lens_' +
+                str(knn_train_samples) + '_knn_train_samples',
+                encoder=embedding_model,
+                family_accessions=knn_test_family_accessions,
+                batch_size=FLAGS.knn_batch_size,
+                train_samples=knn_train_samples,
+                shuffle_seed=FLAGS.knn_shuffle_seed,
+                sample_random_state=FLAGS.knn_sample_random_state))
+
+    encoder_fn_params = None
+    reduce_fn_params = None
+    predict_fn_params = None
+
+    model = create_model(use_transformer=FLAGS.use_transformer,
+                         use_bert=FLAGS.use_bert,
+                         restore_transformer_dir=FLAGS.restore_transformer_dir,
+                         encoder_fn=encoder_fn,
+                         encoder_fn_kwargs=encoder_fn_kwargs,
+                         reduce_fn=reduce_fn,
+                         reduce_fn_kwargs=reduce_fn_kwargs,
+                         layers=layers,
+                         output='prediction',
+                         encoder_fn_params=encoder_fn_params,
+                         reduce_fn_params=reduce_fn_params,
+                         predict_fn_params=predict_fn_params)
+
+    optimizer = create_optimizer(
+        model=model,
+        learning_rate=[FLAGS.encoder_lr, FLAGS.lens_lr, FLAGS.predictor_lr],
+        weight_decay=[FLAGS.encoder_wd, FLAGS.lens_wd, FLAGS.predictor_wd],
+        layers=layers)
+
+    if FLAGS.load_model:
+        optimizer = checkpoints.restore_checkpoint(ckpt_dir=os.path.join(
+            'gs://' + FLAGS.gcs_bucket, FLAGS.load_model_dir),
+                                                   target=optimizer,
+                                                   step=FLAGS.load_model_step)
+
+        trained_params = copy.deepcopy(optimizer.target.params)
+
+        embedding_model = parameter_update(model=embedding_model,
+                                           params=trained_params)
+
+    if FLAGS.save_model:
+        checkpoints.save_checkpoint(ckpt_dir=os.path.join(
+            'gs://' + FLAGS.gcs_bucket, FLAGS.save_model_dir),
+                                    target=optimizer,
+                                    step=FLAGS.load_model_step)
+
+    for i in range(FLAGS.measurements):
+
+        train_batches, train_indexes = create_pfam_batches(
+            family_accessions=lens_knn_train_family_accessions,
+            batch_size=FLAGS.lens_batch_size,
+            samples=FLAGS.lens_train_samples,
+            epochs=measurement_epochs,
+            drop_remainder=True,
+            shuffle_seed=FLAGS.lens_shuffle_seed,
```

Comment:
just realized that it would be better if you shuffled differently each time you called this, but maintained determinism. could do something like shuffle_seed=(FLAGS.lens_shuffle_seed + i).

Reply:
makes sense, should I also change the sample seed?